test/e2e/cache_watch_test.go

```diff
@@ -55,36 +62,50 @@ var _ = Describe("Kubernetes cache watch test:", Label("cache-watch"), func() {
 				}
 			})
 		},
-		Entry("From all namespaces when no namespace config is set", Label("gets-all"),
-			"gets-all", []int{}, []bool{true, true, true}),
-		Entry("One namespace when only one namespace is configured", Label("gets-one"),
-			"gets-one", []int{0}, []bool{true, false, false}),
-		Entry("Two namespaces when only those two are configured", Label("gets-two"),
-			"gets-two", []int{0, 1}, []bool{true, true, false}),
+
+		Entry("From all namespaces when no namespace config is set", Label("gets-all"), &tc{
+			namespaces:      []string{"ns1", "ns2", "ns3"},
+			watchNamespaces: nil,
+			wantToFind:      []string{"ns1", "ns2", "ns3"},
+		}),
+
+		Entry("One namespace when only one namespace is configured", Label("gets-all"), &tc{
+			namespaces:      []string{"ns1", "ns2", "ns3"},
+			watchNamespaces: []string{"ns1"},
+			wantToFind:      []string{"ns1"},
+		}),
+
+		Entry("Two namespaces when only those two are configured", Label("gets-all"), &tc{
```

Comment:
And this `gets-two`

Reply:
indeed, copy-pasta as well, do you mind to fix?

Reply:
Done